weblate/api/tests.py

```diff
@@ -1235,6 +1237,54 @@ def test_patch(self):
         )
         self.assertEqual(response.data["slug"], "new-slug")
 
+    def test_create_component_docfile(self):
+        with open(TEST_DOC, "rb") as handle:
+            response = self.do_request(
+                "api:project-components",
+                self.project_kwargs,
+                method="post",
+                code=201,
+                superuser=True,
+                request={
+                    "docfile": handle,
+                    "name": "Local project",
+                    "slug": "local-project",
+                    "repo": "local:",
+                    "vcs": "local",
+                    "filemask": "*.strings",
+                    "template": "en.strings",
```

Comment:
This is probably why you handle template in the create code. It should not be specified in this case - you have single document you want to translate, not set of files.

```suggestion
                    "filemask": "*.html",
                    "template": "en.html",
```

What would be more user friendly is to use separate serializer for this case and fill in all these values there, but documenting expected parameters in this case is good enough as well.